### ISSUE TITLE
Warning from NSKeyedUnarchiver _warnAboutPlistType:missingInAllowedClasses: in iOS 15

### ISSUE THREAD

**Issue by Westacular** [firebase/firebase-ios-sdk#8276]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: Xcode 13 beta 1
  * Firebase SDK version: 8.1.0
  * Installation method: CocoaPods
  * Firebase Component: GoogleUtilities

#### Steps to reproduce:

It seems Apple is trying to eliminate how NSNumber and NSString have been automatically treated as "safe" / implicitly allowed classes for NSSecureCoding.

In iOS 15 beta 1, there is a new console log warning when NSKeyedUnarchiver unarchives an object of this type and it's _not_ listed in the set of allowed classes for that call:

```
-[NSKeyedUnarchiver _warnAboutPlistType:missingInAllowedClasses:] allowed unarchiving safe plist type ''NSString' (0x1d4c7ef18) [/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/Foundation.framework]', even though it was not explicitly included in the client allowed classes set: '{(
    "'NSDictionary' (0x1d4c5b0a8) [/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/CoreFoundation.framework]",
    "'NSDate' (0x1d4c5aea0) [/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/CoreFoundation.framework]"
)}'. This will be disallowed in the future.
```

The above example is triggered by https://github.com/google/GoogleUtilities/blob/main/GoogleUtilities/Environment/GULHeartbeatDateStorage.m#L140

but there are probably other spots in GoogleUtilities and Firebase that trigger this; you'll need to do some auditing, probably.

**Comment by paulb777** [comment]
Thanks! We appreciate the report and will investigate.

**Comment by stammy** [comment]
I'm experiencing this as well, someone just pointed to it being Firebase after I posted this thread: https://developer.apple.com/forums/thread/683077

**Comment by paulb777** [comment]
We released the fix with GoogleUtilities 7.4.3.  Thanks again for the reports!
